Operate cannot deserialize IN_PROGRESS snapshot state in Opensearch

### <!-- Component -->

<!-- Operate- -->

### <!-- Affected version -->

<!-- -8.8 -->, <!-- -8.7 -->, <!-- -8.6 -->, <!-- -8.5 -->, <!-- -8.4 -->

### <!-- Severity -->

<!-- High- -->

### <!-- Likelihood -->

<!-- Low_ -->

### Description

Operate is not able to deserialize `IN_PROGRESS` snapshot state returned by Opensearch when polling for backup status. As a result, calls to Get Backup status throws an exception whenever a snapshot is still in progress.

The list of supported snapshot state by Opensearch are documented here https://docs.opensearch.org/docs/latest/api-reference/snapshots/get-snapshot-status/#snapshot-states

### Steps to reproduce

It is hard to reproduce as this requires having a snapshot in IN_PROGRESS state in OpenSearch

### Current behavior

```
io.camunda.operate.webapp.api.v1.exceptions.ClientException: Exception occurred when searching for backup with ID [...].
	at io.camunda.operate.webapp.api.v1.rest.ErrorController.handleException(ErrorController.java:66) ~[operate-webapp-8.6.9.jar:8.6.9]
...
Caused by: io.camunda.operate.exceptions.OperateRuntimeException: Exception occurred when searching for backup with ID [...].
	at io.camunda.operate.webapp.opensearch.backup.OpensearchBackupRepository.findSnapshots(OpensearchBackupRepository.java:342) ~[operate-webapp-8.6.9.jar:8.6.9]
	at io.camunda.operate.webapp.opensearch.backup.OpensearchBackupRepository.getBackupState(OpensearchBackupRepository.java:140) ~[operate-webapp-8.6.9.jar:8.6.9]
	at io.camunda.operate.webapp.backup.BackupService.getBackupState(BackupService.java:190) ~[operate-webapp-8.6.9.jar:8.6.9]
	at io.camunda.operate.webapp.management.BackupController.getBackupState(BackupController.java:56) ~[operate-webapp-8.6.9.jar:8.6.9]
...
Caused by: java.lang.IllegalArgumentException: No enum constant io.camunda.operate.store.opensearch.response.SnapshotState.IN_PROGRESS
	at java.base/java.lang.Enum.valueOf(Enum.java:293) ~[?:?]
	at io.camunda.operate.store.opensearch.response.SnapshotState.valueOf(SnapshotState.java:10) ~[operate-schema-8.6.9.jar:8.6.9]
	at io.camunda.operate.store.opensearch.client.sync.OpenSearchSnapshotOperations.mapToSnapshotInfo(OpenSearchSnapshotOperations.java:82) ~[operate-schema-8.6.9.jar:8.6.9]
```

### Expected behavior

`IN_PROGRESS` state deserialized with no errors

### Environment

SM

### Version

- Component Version: 8.6.9


### Rootcause

_No response_

### Solution ideas

_No response_

### Dev -> QA handover

- Resources:
- Versions to validate:
- Release version (in which version this feature will be released):


### Links

_No response_

--- a/operate/schema/src/main/java/io/camunda/operate/store/opensearch/response/SnapshotState.java
+++ b/operate/schema/src/main/java/io/camunda/operate/store/opensearch/response/SnapshotState.java
@@ -8,18 +8,8 @@
 package io.camunda.operate.store.opensearch.response;
 
 public enum SnapshotState {
-  FAILED("FAILED"),
-  PARTIAL("PARTIAL"),
-  STARTED("STARTED"),
-  SUCCESS("SUCCESS");
-  private final String state;
-
-  SnapshotState(final String state) {
-    this.state = state;
-  }
-
-  @Override
-  public String toString() {
-    return state;
-  }
+  FAILED,
+  PARTIAL,
+  IN_PROGRESS,
+  SUCCESS;
 }

--- a/operate/schema/src/test/java/io/camunda/operate/schema/store/opensearch/client/sync/OpenSearchSnapshotOperationsTest.java
+++ b/operate/schema/src/test/java/io/camunda/operate/schema/store/opensearch/client/sync/OpenSearchSnapshotOperationsTest.java
@@ -98,7 +98,7 @@ class OpenSearchSnapshotOperationsTest {
                     "uuid",
                     "uuid-value",
                     "state",
-                    "STARTED",
+                    "IN_PROGRESS",
                     "start_time_in_millis",
                     23L,
                     "metadata",
@@ -136,7 +136,7 @@ class OpenSearchSnapshotOperationsTest {
   }
 
   private void assertSnapshotInfo(final OpenSearchSnapshotInfo snapshotInfo) {
-    assertThat(snapshotInfo.getState()).isEqualTo(SnapshotState.STARTED);
+    assertThat(snapshotInfo.getState()).isEqualTo(SnapshotState.IN_PROGRESS);
     assertThat(snapshotInfo.getStartTimeInMillis()).isEqualTo(23L);
     assertThat(snapshotInfo.getMetadata()).isEmpty();
     assertThat(snapshotInfo.getFailures()).isEmpty();

--- a/operate/webapp/src/main/java/io/camunda/operate/webapp/opensearch/backup/OpensearchBackupRepository.java
+++ b/operate/webapp/src/main/java/io/camunda/operate/webapp/opensearch/backup/OpensearchBackupRepository.java
@@ -262,7 +262,7 @@ public class OpensearchBackupRepository implements BackupRepository {
                     // No need to continue
                     onFailure.run();
                     break;
-                  } else if (STARTED.equals(maybeCurrentSnapshot.get().getState())) {
+                  } else if (IN_PROGRESS.equals(maybeCurrentSnapshot.get().getState())) {
                     ThreadUtil.sleepFor(100);
                   } else {
                     handleSnapshotReceived(maybeCurrentSnapshot.get(), onSuccess, onFailure);
@@ -370,7 +370,9 @@ public class OpensearchBackupRepository implements BackupRepository {
         .map(OpenSearchSnapshotInfo::getState)
         .anyMatch(s -> FAILED.equals(s) || PARTIAL.equals(s))) {
       return BackupStateDto.FAILED;
-    } else if (snapshots.stream().map(OpenSearchSnapshotInfo::getState).anyMatch(STARTED::equals)) {
+    } else if (snapshots.stream()
+        .map(OpenSearchSnapshotInfo::getState)
+        .anyMatch(IN_PROGRESS::equals)) {
       return BackupStateDto.IN_PROGRESS;
     } else if (snapshots.size() < expectedSnapshotsCount) {
       return BackupStateDto.INCOMPLETE;

--- a/operate/webapp/src/test/java/io/camunda/operate/webapp/opensearch/backup/OpensearchBackupRepositoryTest.java
+++ b/operate/webapp/src/test/java/io/camunda/operate/webapp/opensearch/backup/OpensearchBackupRepositoryTest.java
@@ -93,7 +93,7 @@ class OpensearchBackupRepositoryTest {
         List.of(
             new OpenSearchSnapshotInfo()
                 .setSnapshot("test-snapshot")
-                .setState(SnapshotState.STARTED)
+                .setState(SnapshotState.IN_PROGRESS)
                 .setStartTimeInMillis(23L));
     final var response = new OpenSearchGetSnapshotResponse(snapshotInfos);
 
@@ -125,7 +125,7 @@ class OpensearchBackupRepositoryTest {
         List.of(
             new OpenSearchSnapshotInfo()
                 .setSnapshot("test-snapshot")
-                .setState(SnapshotState.STARTED));
+                .setState(SnapshotState.IN_PROGRESS));
     final var response = new OpenSearchGetSnapshotResponse(snapshotInfos);
     mockObjectMapperForMetadata(metadata);
     when(openSearchSnapshotOperations.get(any())).thenReturn(response);
@@ -151,7 +151,7 @@ class OpensearchBackupRepositoryTest {
         List.of(
             new OpenSearchSnapshotInfo()
                 .setSnapshot("test-snapshot")
-                .setState(SnapshotState.STARTED));
+                .setState(SnapshotState.IN_PROGRESS));
     final var response = new OpenSearchGetSnapshotResponse(snapshotInfos);
     mockObjectMapperForMetadata(metadata);
     when(openSearchSnapshotOperations.get(any())).thenReturn(response);

--- a/operate/webapp/src/test/java/io/camunda/operate/webapp/opensearch/backup/OpensearchBackupRepositoryTest.java
+++ b/operate/webapp/src/test/java/io/camunda/operate/webapp/opensearch/backup/OpensearchBackupRepositoryTest.java
@@ -112,7 +112,7 @@ class OpensearchBackupRepositoryTest {
     assertThat(snapshotDtoDetails).hasSize(1);
     final var snapshotDtoDetail = snapshotDtoDetails.get(0);
     assertThat(snapshotDtoDetail.getSnapshotName()).isEqualTo("test-snapshot");
-    assertThat(snapshotDtoDetail.getState()).isEqualTo("STARTED");
+    assertThat(snapshotDtoDetail.getState()).isEqualTo("IN_PROGRESS");
     assertThat(snapshotDtoDetail.getFailures()).isNull();
     assertThat(snapshotDtoDetail.getStartTime().toInstant().toEpochMilli()).isEqualTo(23L);
   }

